### PR TITLE
gtkdoc: Add 'check' kwarg

### PR DESCRIPTION
**Pull request by xclaesse**
This runs gtkdoc-check in meson tests.

### CODE REVIEW

File: mesonbuild/modules/gnome.py [hunk 1]
```diff
@@ -1022,6 +1023,16 @@ def gtkdoc(self, state, args, kwargs):
         res = [build.RunTarget(targetname, command[0], command[1:] + args, depends, state.subdir, state.subproject)]
         if kwargs.get('install', True):
             res.append(build.RunScript(command, args))
+        if kwargs.get('check', False):
+            check_cmd = self.interpreter.find_program_impl('gtkdoc-check')
+            check_env = ['DOC_MODULE=' + modulename,
+                         'DOC_MAIN_SGML_FILE=' + main_file]
+            check_args = [targetname + '-check', check_cmd]
+            check_kwargs = {'env' : check_env,
```
Comment (ignatenkobrain) [inline mesonbuild/modules/gnome.py]: __[Flake8]__ [E203] whitespace before ':' ([link](https://sider.review/gh/repos/19784232/pulls/5591#issue-4295978))

<sub>posted by Sider</sub>

File: mesonbuild/modules/gnome.py [hunk 2]
```diff
@@ -1022,6 +1023,16 @@ def gtkdoc(self, state, args, kwargs):
...
+                            'workdir' : os.path.join(state.environment.get_build_dir(), state.subdir)}
```
Comment (ignatenkobrain) [inline mesonbuild/modules/gnome.py]: __[Flake8]__ [E203] whitespace before ':' ([link](https://sider.review/gh/repos/19784232/pulls/5591#issue-4295979))

<sub>posted by Sider</sub>

**Comment by ocrete** [comment]
In libnice, I also need to set SRCDIR=... and BUILDDIR=... in the env when running it

**Comment by xclaesse** [comment]
> In libnice, I also need to set SRCDIR=... and BUILDDIR=... in the env when running it

Are you sure? Why would it need that? Checking the code of gtkdoc-check and I don't see it using those env.

**Comment by nirbheek** [comment]
Does this PR contain the `alias_target()` commit on purpose?

**Comment by xclaesse** [comment]
> Does this PR contain the `alias_target()` commit on purpose?

Yes, I'm depending on it, but PR https://github.com/mesonbuild/meson/pull/5606 can be merged first.

**Comment by xclaesse** [comment]
`alias_target()` PR has been merged, so this PR is ready to be reviewed now.

**Comment by dcbaker** [comment]
Probably should have a release snippet as well?

**Comment by xclaesse** [comment]
> Probably should have a release snippet as well?

Good point, done.

**Comment by jpakkane** [comment]
LGTM, I'll merge this once CI gives the green light.

**Pull request merged**